src/main/java/com/example/sharemind/consult/repository/ConsultRepository.java

```diff
@@ -1,9 +1,17 @@
 package com.example.sharemind.consult.repository;
 
 import com.example.sharemind.consult.domain.Consult;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ConsultRepository extends JpaRepository<Consult, Long> {
+    @Query("SELECT realtime.realtimeId FROM Consult c JOIN c.realtimeConsult realtime WHERE c.customer.customerId = :customerId")
+    List<Long> findRealtimeConsultIdsByCustomerId(Long customerId); //todo: 쿼리 최적화 필요
```

Comment:
전 @ Query를 거의 사용하지 않는 편이라 쿼리 최적화는 잘 모르지만 저라면 어떻게 했을까를 생각해봤습니다...!
```suggestion
    List<Consult> findAllByCustomerAndRealtimeConsultIsNotNull(Customer customer);
    
    // ConsultServiceImpl
    public List<RealtimeConsult> getRealtimeConsultsByCustomer(Customer customer) {
         return consultRepository.findAllByCustomerAndRealtimeConsultIsNotNull(customer).stream()
              .map(consult -> consult.getRealtimeConsult())
              .toList();
    }
    
    // RealtimeConsultServiceImpl
    Customer customer = customerService.getCustomerByCustomerId(userId);
    return consultService.getRealtimeConsultsByCustomer(customer).stream()
         .map(realtimeConsult -> realtimeConsult.getRealtimeConsultId())
         .toList();
```

Reply:
감사합니다~~!